woocommerce/google-listings-and-ads
# Setup Merchant Center Step 3 - Auto Save and Load Data with mc/settings API

### Changes proposed in this Pull Request:

(Note that this PR contains changes from https://github.com/woocommerce/google-listings-and-ads/pull/138. Please review and merge https://github.com/woocommerce/google-listings-and-ads/pull/138 first before this one.)

Closes #157 .

This PR will automatically save and load the form data for the Setup Merchant Center - Step 3 page. When the pre-launch checklist is completed, it will fire `gla_pre_launch_checklist_complete` track event.

This PR uses the API created from https://github.com/woocommerce/google-listings-and-ads/pull/137:

> Options added
> 
> * shipping_rate (flat or manual)
> * offers_free_shipping (bool)
> * free_shipping_threshold (int)
> * shipping_time (flat or manual)
> * offers_free_shipping (bool)
> * share_shipping_time (bool)
> * tax_rate (destination or manual)
> 
> Existing options (for reference) relating to shopping policies include
> 
> * website_live (bool)
> * checkout_process_secure (bool)
> * payment_methods_visible (bool)
> * refund_tos_visible (bool)
> * contact_info_visible (bool)

### Screenshots:

Demo video with my voice: https://d.pr/v/nZGhsW

### Detailed test instructions:

1. Open https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc
2. Click the radio buttons and checkboxes.
3. Refresh the browser. 
4. The previously entered data should be automatically loaded back.
5. Complete the prelaunch checklist. The `gla_pre_launch_checklist_complete` track event should be called.
